From apache@plt-scheme.org Mon Apr  5 22:49:24 2004
Received: from champlain.ccs.neu.edu (champlain.ccs.neu.edu [127.0.0.1])
	by champlain.ccs.neu.edu (8.12.10/8.12.10) with ESMTP id i362nODU008573
	for <bugs@plt-scheme.org>; Mon, 5 Apr 2004 22:49:24 -0400
Message-Id: <200404060249.i362nOvY008571@champlain.ccs.neu.edu>
Date: Mon, 5 Apr 2004 22:49:24 -0400
From: dsilva@ccs.neu.edu
To: bugs@plt-scheme.org
Subject: module name resovler doesn't like symlinks

>Number:         6800
>Category:       mzscheme
>Synopsis:       module name resovler doesn't like symlinks
>Class:          sw-bug
>Responsible:    mflatt
>Severity:       serious
>Priority:       medium
>State:          open
>Confidential:   no
>Arrival-Date:   Mon Apr 05 22:50:00 -0400 2004
>Last-Modified:  Sun May 02 13:03:23 -0400 2004
>Originator:     Daniel Silva
>Organization:
plt
>Submitter-Id:   unknown
>Release:        206.1
>Environment:
linux-other
>Description:
I've received reports that when a collection in ~/.plt-scheme/206p1/collects/<collect> is a symlink to some o
ther folder, libraries don't get loaded correctly with (lib "modulename" "collection-name")

I received this through email:
"""
I tried "mzscheme --require python.ss".  I think somewhere around this point I figured out that I needed to d
o something like "mkdir -p ~/.plt-scheme/206p1/collects; ln -s ~/Projects/plt/collects/python ~/.plt-scheme/2
06p1/collects/".
Then once I successfully ran the command, I got tons of debugging output, and then:

 > (py-eval "1")
input string:1:0: require: broken compiled code (phase 0): cannot find module |,/home/radix/Projects/plt/coll
ects/python/c-bindings| in: make-py-number
"""

py-eval compiles and evaluates code in an environment initialized with this function:

  (define init-python-namespace
    (let ()
    (dynamic-require '(lib "base.ss" "python") #f)
    (let ([base-path ((current-module-name-resolver) '(lib "base.ss" "python") #f
#f)]
          [outer-namespace (current-namespace)])
      (lambda (py-ns)
          (parameterize ([current-namespace py-ns])
            (namespace-attach-module outer-namespace base-path)
            (namespace-require base-path))))))

>How-To-Repeat:
>Fix:
>Audit-Trail:
Responsible changed from "nobody" to "mflatt" by eli at Sun, 02 May 2004 13:03:23 -0400
Reason>>> mzscheme

